### PR TITLE
[Key Vault] Authentication documentation updates

### PR DESCRIPTION
**Pull request by mccoyp**
# Description

Resolves https://github.com/Azure/azure-sdk-for-python/issues/25039, based on https://github.com/Azure/azure-sdk-for-java/pull/29995 and https://github.com/Azure/azure-sdk-for-js/pull/22814. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by vcolin7 — APPROVED** [review]

File: sdk/keyvault/azure-keyvault-certificates/README.md [hunk 1]
```diff
@@ -29,102 +30,15 @@ authentication as demonstrated below.
...
-#### Create a service principal (optional)
-This [Azure Cloud Shell][azure_cloud_shell] snippet shows how to create a
-new service principal. Before using it, replace "your-application-name" with
-a more appropriate name for your service principal.
-
-Create a service principal:
-```Bash
-az ad sp create-for-rbac --name http://my-application --skip-assignment
-```
+In order to interact with the Azure Key Vault service, you will need an instance of a [CertificateClient][certificate_client_docs], as well as a **vault url** (which you may see as "DNS Name" in the Azure Portal) and a credential object. This document demonstrates using a [DefaultAzureCredential][default_cred_ref], which is appropriate for most scenarios, including local development and production environments. We recommend using a [managed identity][managed_identity] for authentication in production environments.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/README.md:36]: Does the vault URL appear as DNS Name for Key Vault as well? I only had that for Managed HSM based on some docs I copied from .NET a while ago.

File: sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py [hunk 2]
```diff
@@ -6,16 +6,16 @@
...
 
 # ----------------------------------------------------------------------------------------------------------
 # Prerequisites:
 # 1. An Azure Key Vault (https://docs.microsoft.com/en-us/azure/key-vault/quick-create-cli)
 #
 # 2. azure-keyvault-certificates and azure-identity packages (pip install these)
 #
-# 3. Set Environment variables AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET, VAULT_URL
-#    (See https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/keyvault/azure-keyvault-keys#authenticate-the-client)
+# 3. Set up your environment to use azure-identity's DefaultAzureCredential. To authenticate a service principal with
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py:16]: We are trying to not direct customers to using client credentials for authentication, you might want to rephrase to something like [this](https://github.com/Azure/azure-sdk-for-java/blob/8b31a1f08916c8b8089aa756710f1ebc75e98c8d/sdk/keyvault/azure-security-keyvault-certificates/src/samples/java/com/azure/security/keyvault/certificates/BackupAndRestoreOperations.java#L42). There's a bunch of files in the Certificates samples doing the same.

File: sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py [hunk 3]
```diff
@@ -31,10 +31,8 @@
 # 5. Restore a certificate (restore_certificate_backup)
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py:34]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations_async.py [hunk 4]
```diff
@@ -34,10 +34,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations_async.py:37]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/contacts.py [hunk 5]
```diff
@@ -26,10 +26,8 @@
 # 3. Delete contacts (delete_contacts)
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/contacts.py:29]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/contacts_async.py [hunk 6]
```diff
@@ -30,10 +30,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/contacts_async.py:33]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/hello_world.py [hunk 7]
```diff
@@ -29,10 +29,8 @@
 #
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/hello_world.py:32]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/hello_world_async.py [hunk 8]
```diff
@@ -33,10 +33,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/hello_world_async.py:36]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/import_certificate_async.py [hunk 9]
```diff
@@ -33,7 +33,7 @@
 # ----------------------------------------------------------------------------------------------------------
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/import_certificate_async.py:36]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/issuers.py [hunk 10]
```diff
@@ -30,10 +30,8 @@
 # 5. Delete an issuer (delete_issuer)
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/issuers.py:33]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/issuers_async.py [hunk 11]
```diff
@@ -34,10 +34,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/issuers_async.py:37]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/list_operations.py [hunk 12]
```diff
@@ -32,9 +31,8 @@
 #
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service. Notice that the client is using default
-# Azure credentials. To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/list_operations.py:34]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/list_operations_async.py [hunk 13]
```diff
@@ -34,9 +34,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service. Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/list_operations_async.py:37]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/parse_certificate.py [hunk 14]
```diff
@@ -36,10 +36,8 @@
 #
 # ----------------------------------------------------------------------------------------------------------
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/parse_certificate.py:39]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/parse_certificate_async.py [hunk 15]
```diff
@@ -39,10 +39,8 @@
 # ----------------------------------------------------------------------------------------------------------
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/parse_certificate_async.py:42]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/recover_purge_operations.py [hunk 16]
```diff
@@ -30,10 +30,8 @@
 # ----------------------------------------------------------------------------------------------------------
 
 
-# Instantiate a certificate client that will be used to call the service.
-# Notice that the client is using default Azure credentials.
-# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-# 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+# Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/recover_purge_operations.py:33]:
```suggestion
# Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-certificates/samples/recover_purge_operations_async.py [hunk 17]
```diff
@@ -32,10 +32,8 @@
 
 
 async def run_sample():
-    # Instantiate a certificate client that will be used to call the service.
-    # Notice that the client is using default Azure credentials.
-    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
-    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    # Instantiate a backup client that will be used to call the service.
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-certificates/samples/recover_purge_operations_async.py:35]:
```suggestion
    # Instantiate a certificate client that will be used to call the service.
```

File: sdk/keyvault/azure-keyvault-keys/README.md [hunk 18]
```diff
@@ -31,105 +32,15 @@ authentication as demonstrated below.
...
-          "tenantId": "...",
-          "vaultUri": "https://my-key-vault.vault.azure.net/"
-      },
-      "resourceGroup": "my-resource-group",
-      "type": "Microsoft.KeyVault/vaults"
-  }
-  ```
-
-  > The `"vaultUri"` property is the `vault_url` used by [KeyClient][key_client_docs]
+* An existing [Azure Key Vault][azure_keyvault]. If you need to create one, you can do so using the Azure CLI by following the steps in [this document][azure_keyvault_cli].
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-keys/README.md:35]: We should also mention they can use a Managed HSM.

File: sdk/keyvault/azure-keyvault-keys/samples/README.md [hunk 19]
```diff
@@ -10,19 +10,32 @@ urlFragment: keyvault-keys-samples
...
-* [recover_purge_operations.py][recover_purge_sample] and
-[recover_purge_operations_async.py][recover_purge_async_sample] - recover and purge keys
-* [key_rotation.py][key_rotation_sample] and
-[key_rotation_async.py][key_rotation_async_sample] - rotate keys automatically and on-demand
+## Prerequisites
+
+You must have an [Azure subscription](https://azure.microsoft.com/free) and an
+[Azure Key Vault](https://docs.microsoft.com/azure/key-vault/general/overview) to run
+these samples. You can create a key vault with the
+[Azure CLI](https://docs.microsoft.com/azure/key-vault/general/quick-create-cli).
```
Comment (vcolin7) [inline sdk/keyvault/azure-keyvault-keys/samples/README.md:18]: We should also mention they can use a Managed HSM.

File: sdk/keyvault/azure-keyvault-certificates/README.md (hunk 1 above)
Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-certificates/README.md:36]: Good point; it doesn't appear to. Fixed!

**Review by vcolin7 — APPROVED** [review]

**Comment by mccoyp** [comment]
CI still broken because of unrelated pylint bug that's being fixed by APIView team and EngSys. There aren't any errors introduced in cspell, pylint, or the broken link checks, so I'll override check enforcer for this PR.

**Pull request merged**